### PR TITLE
Improved W&B migration script

### PR DESCRIPTION
**Pull request by SiddhantSadangi**
# Description

* Improved performance by >6x
* Improved README with benchmark and post-migration information

__Related to:__ Improve W&B migration script

__Any expected test failures?__

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [X] adds a new feature
- [ ] fixes breaking code
- [X] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [X] Refactored code ([sourcery](https://sourcery.ai/))
- [X] Tested code locally
- [X] Precommit installed and run before pushing changes
- [ ] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [ ] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS:
- Python version: 3.10.11
- Neptune version: 1.10.4
- Affected libraries with version: wandb==0.17.4

### CODE REVIEW

File: utils/migration_tools/from_wandb/README.md [hunk 1]
```diff
@@ -1,64 +1,83 @@
...
-|    System logs    |          output.log          |          run.monitoring.stdout           |
-|     Notebook      | code/_session_history.ipynb  |          run.source_code.files           |
-| requirements.txt  |       requirements.txt       |          run.source_code.files           |
-| Model checkpoints |    \*.ckpt/\*checkpoint\*    |             run.checkpoints              |
-|    Other files    |         run.files()          |                run.files                 |
+## Instructions
+
+To use the script, follow these steps:
+
+1. Run `wandb_to_neptune.py`
```
Comment (szaganek) [inline utils/migration_tools/from_wandb/README.md:13]:
```suggestion
1. Run `wandb_to_neptune.py`.
```

File: utils/migration_tools/from_wandb/README.md [hunk 2]
```diff
@@ -1,64 +1,83 @@
...
+2. Enter the source W&B entity name. Leave blank to use your default entity
```
Comment (szaganek) [inline utils/migration_tools/from_wandb/README.md:14]:
```suggestion
2. Enter the source W&B entity name. Leave blank to use your default entity.
```

File: utils/migration_tools/from_wandb/README.md [hunk 3]
```diff
@@ -1,64 +1,83 @@
...
+3 Enter the destination Neptune workspace name. Leave blank to read from the `NEPTUNE_PROJECT` environment variable
```
Comment (szaganek) [inline utils/migration_tools/from_wandb/README.md:15]:
```suggestion
3. Enter your Neptune project name in the form `workspace-name/project-name`. Leave blank to read from the `NEPTUNE_PROJECT` environment variable.
```
It's not just the workspace name, right?

File: utils/migration_tools/from_wandb/README.md [hunk 4]
```diff
@@ -1,64 +1,83 @@
...
+3. Enter the number of workers to use to copy the metadata. Leave blank to let `ThreadPoolExecutor` decide
```
Comment (szaganek) [inline utils/migration_tools/from_wandb/README.md:16]:
```suggestion
3. Enter the number of workers to use to copy the metadata. Leave blank to let `ThreadPoolExecutor` decide.
```

File: utils/migration_tools/from_wandb/README.md (hunk 3 above)
Comment (SiddhantSadangi) [reply utils/migration_tools/from_wandb/README.md:15]: It is. Projects are created automatically

File: utils/migration_tools/from_wandb/README.md [hunk 5]
```diff
@@ -1,64 +1,83 @@
...
-| requirements.txt  |       requirements.txt       |          run.source_code.files           |
-| Model checkpoints |    \*.ckpt/\*checkpoint\*    |             run.checkpoints              |
-|    Other files    |         run.files()          |                run.files                 |
+## Instructions
+
+To use the script, follow these steps:
+
+1. Run `wandb_to_neptune.py`.
+2. Enter the source W&B entity name. Leave blank to use your default entity.
+3 Enter the destination Neptune workspace name. Leave blank to read from the `NEPTUNE_PROJECT` environment variable
```
Comment (szaganek) [inline utils/migration_tools/from_wandb/README.md:15]:
```suggestion
3. Enter the destination Neptune workspace name. Leave blank to read from the `NEPTUNE_PROJECT` environment variable.
```
Plus, the following step is also `3.`

File: utils/migration_tools/from_wandb/README.md [hunk 6]
```diff
@@ -1,64 +1,83 @@
...
-| requirements.txt  |       requirements.txt       |          run.source_code.files           |
-| Model checkpoints |    \*.ckpt/\*checkpoint\*    |             run.checkpoints              |
-|    Other files    |         run.files()          |                run.files                 |
+## Instructions
+
+To use the script, follow these steps:
+
+1. Run `wandb_to_neptune.py`.
+1. Enter the source W&B entity name. Leave blank to use your default entity.
+1. Enter the destination Neptune workspace name. Leave blank to read from the `NEPTUNE_PROJECT` environment variable
```
Comment (szaganek) [inline utils/migration_tools/from_wandb/README.md:15]:
```suggestion
1. Enter the destination Neptune workspace name. Leave blank to read from the `NEPTUNE_PROJECT` environment variable.
```

**Review by szaganek — DISMISSED** [review]

**Pull request merged**